pkg/services/publicdashboards/api/api.go

```diff
@@ -157,35 +151,16 @@ func (api *Api) QueryPublicDashboard(c *models.ReqContext) response.Response {
 		return response.Error(http.StatusBadRequest, "invalid panel ID", err)
 	}
 
-	// Get the dashboard
-	pubdash, dashboard, err := api.PublicDashboardService.GetPublicDashboard(c.Req.Context(), web.Params(c.Req)[":accessToken"])
-	if err != nil {
-		return response.Error(http.StatusInternalServerError, "could not fetch dashboard", err)
-	}
-
-	// Build the request data objecct
-	reqDTO, err := api.PublicDashboardService.BuildPublicDashboardMetricRequest(
-		c.Req.Context(),
-		dashboard,
-		pubdash,
-		panelId,
-	)
-	if err != nil {
-		return handleDashboardErr(http.StatusInternalServerError, "Failed to get queries for public dashboard", err)
+	reqDTO := &PublicDashboardQueryDTO{}
+	if err = web.Bind(c.Req, reqDTO); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
 
-	// Build anonymous user for the request
-	anonymousUser, err := api.PublicDashboardService.BuildAnonymousUser(c.Req.Context(), dashboard)
+	resp, err := api.PublicDashboardService.GetQueryDataResponse(c.Req.Context(), c.SkipCache, reqDTO, panelId, web.Params(c.Req)[":accessToken"])
```

Comment:
nit/thought: Now that we're only calling this one method `GetQueryDataResponse` I wonder if there is a better name? `QueryData` or something.